internal/workflow/imageworkflow/getlatestversion.go

```diff
@@ -20,6 +22,65 @@ func GetLatestReference() workflow.Step {
 				return nil, err
 			}
 
+			// If the tag is not adhering to semver, try to identify whether or not
```

Comment:
Should we switch the order of operations?

1. Check semver and update the tag
2. Check manifest digest

This will catch the case of 8.0 being in use, with no new 8.1. So no new semver. But 8.0 can still have been changed. I don't see any point of enforcing when it's done. We always fetch the latest manifest anyway?

Reply:
It could open up an issue if we've misunderstood the digest in use - causing us to complain about an update for 8.0 that doesn't exist, just because we got another manifest. I _think_ it should be fine in most cases. It seems to work well for a wide variety of images and registries.

Reply:
I've reworked this logic a bit to what I was getting at in the comments above.